### PR TITLE
Fixing API Calls

### PR DESCRIPTION
**Pull request by pedroserano**
"CURR" is not valid for light brightness and fan speed, the call is "ACTUAL". For fan direction in reverse, it's "REV", rather than "BKW".

### CODE REVIEW

**Comment by pedroserano** [comment]
It was very helpful to find out that I could use telnet to see exactly what commands were sent to the fan. I used the IP address of the fan and the command line: "telnet (insert ip address of the fan) 31415"

File: BigAssApi.js [hunk 1]
```diff
@@ -196,31 +216,35 @@ function BigAssFan (name, id, address, master) {
...
-    this.fan.createGetField('speed', ['FAN', 'SPD'], true, 'CURR');
+    this.fan.createGetField('speed', ['FAN', 'SPD'], true, 'ACTUAL'); //0-7
     this.fan.createGetField('min', ['FAN', 'SPD'], true, 'MIN');
     this.fan.createGetField('max', ['FAN', 'SPD'], true, 'MAX');
-    this.fan.createGetField('auto', ['FAN', 'AUTO'], true, undefined);
-    this.fan.createGetField('whoosh', ['FAN', 'WHOOSH'], true, "STATUS");
-    this.fan.createGetField('isSpinningForwards', ['FAN', 'DIR'], true, undefined, "FWD", "BKW"); // TODO: Check backwards string...
+    this.fan.createGetField('auto', ['FAN', 'AUTO'], true, undefined, "ON", "OFF"); // fan sensor enabled
+    this.fan.createGetField('whoosh', ['FAN', 'WHOOSH'], true, "STATUS"); // ON, OFF
+    this.fan.createGetField('isSpinningForwards', ['FAN', 'DIR'], true, undefined, "FWD", "REV");
```
Comment (sean9keenan) [inline BigAssApi.js:208]: Thanks for fixing this!

File: BigAssApi.js [hunk 2]
```diff
@@ -196,31 +216,35 @@ function BigAssFan (name, id, address, master) {
...
+    this.sensor.createGetField('smartmodestate', ['SMARTMODE', 'STATE'], true, undefined, 'LEARN', 'COOLING', 'HEATING', 'FOLLOWSTAT'); //FOLLOWSTAT is the works with nest option, it is followed by SMARTMODE;ACTUAL;OFF command
 
     this.learn = new BigAssProperty('learn', this);
-    this.learn.createGetField('isOn', ['LEARN', 'STATE'], true, undefined, 'ON', 'OFF');
+    this.learn.createGetField('isOn', ['LEARN', 'STATE'], true, undefined, 'LEARN', 'OFF'); // LEARN appears to be the on command rather than ON, ie LEARN;STATE;LEARN. When turned on, two or three commands follow, WINTERMODE;STATE;OFF and SMARTMODE;STATE;COOLING and SMARTMODE;ACTUAL;COOLING
     this.learn.createGetField('minSpeed', ['LEARN', 'MINSPEED'], true);
     this.learn.createGetField('maxSpeed', ['LEARN', 'MAXSPEED'], true);
-    this.learn.createGetField('zeroTemp', ['LEARN', 'ZEROTEMP'], true); // ??? Wat.
+    this.learn.createGetField('zeroTemp', ['LEARN', 'ZEROTEMP'], true); // this is a four digit number that represents the temperature in celsius (without a decimal) at which the fan automatically turns off in smart mode. For instance '2111' is 21.11 C which is 70 F
```
Comment (sean9keenan) [inline BigAssApi.js:231]: 👍

File: BigAssApi.js [hunk 3]
```diff
@@ -60,6 +60,26 @@ function FanMaster (numberOfExpectedFans) {
             this.rescanForFans();
         }
     }.bind(this)
+    */
+        handleNewFan = function(msg, address) {
+        if (msg[0] == "ALL") {
+            return; // Message not addressed to us
+        }
+        if (msg[4] == "FAN,LSERIES") { //this is less than perfect, but at least allows a program to continue passed the onfanfullyupdated function when a wall control is present - obviously needs to not be LSERIES specfic - need a function to just look at the "FAN" part of that string
```
Comment (sean9keenan) [inline BigAssApi.js]: Hmm, for these messages - what's the full message that is found from the fan? Unfortunately I think this breaks non-LSeries fans. If we want to get just the FAN part we can use do:

```
var onlyFanPortion = split(msg[4], ",")[0] // Grab first part of string before ","
if (onlyFanPortion == "FAN") {
...
```

File: BigAssApi.js [hunk 4]
```diff
@@ -60,6 +60,26 @@ function FanMaster (numberOfExpectedFans) {
             this.rescanForFans();
         }
     }.bind(this)
+    */
+        handleNewFan = function(msg, address) {
```
Comment (sean9keenan) [inline BigAssApi.js:47]: Indentation - also I'd rather you just remove+replace the code you commented out

File: BigAssApi.js [hunk 5]
```diff
@@ -60,6 +60,26 @@ function FanMaster (numberOfExpectedFans) {
...
+            	var newFan = new BigAssFan(msg[0], msg[3], address, this);
+            	this.allFans[msg[0]] = newFan;
+            	this.onFanConnection(newFan);
+            	newFan.updateAll(function() {
+                	this.onFanFullyUpdated(newFan)
+            	}.bind(this));
+            }
+        if (msg[4] == "SWITCH,SENSEME") {
+            myLogWrapper("skipping wall control for now")}
+         else {
```
Comment (sean9keenan) [inline BigAssApi.js]: Extra space before else

File: BigAssApi.js [hunk 6]
```diff
@@ -60,6 +60,26 @@ function FanMaster (numberOfExpectedFans) {
...
+        handleNewFan = function(msg, address) {
+        if (msg[0] == "ALL") {
+            return; // Message not addressed to us
+        }
+        if (msg[4] == "FAN,LSERIES") { //this is less than perfect, but at least allows a program to continue passed the onfanfullyupdated function when a wall control is present - obviously needs to not be LSERIES specfic - need a function to just look at the "FAN" part of that string
+            	var newFan = new BigAssFan(msg[0], msg[3], address, this);
+            	this.allFans[msg[0]] = newFan;
+            	this.onFanConnection(newFan);
+            	newFan.updateAll(function() {
+                	this.onFanFullyUpdated(newFan)
```
Comment (sean9keenan) [inline BigAssApi.js:57]: Really far indented now

File: BigAssApi.js [hunk 7]
```diff
@@ -196,31 +216,35 @@ function BigAssFan (name, id, address, master) {
 
     this.fan = new BigAssProperty('fan', this);
     this.fan.createGetField('isOn', ['FAN', 'PWR'], true, undefined, "ON", "OFF");
-    this.fan.createGetField('speed', ['FAN', 'SPD'], true, 'CURR');
+    this.fan.createGetField('speed', ['FAN', 'SPD'], true, 'ACTUAL'); //0-7
```
Comment (sean9keenan) [inline BigAssApi.js:203]: I'm going to have to check that this is the same for the Haiku series. I doubt they would change this - but nothing's impossible!

**Comment by sean9keenan** [comment]
Thanks for the additional comments and help fixing up some of the issues!

I have a few comments - mostly formatting related, and then I would be fine to merge it in. Nice to see someone helping out and using the tool!

File: BigAssApi.js [hunk 8]
```diff
@@ -60,6 +60,26 @@ function FanMaster (numberOfExpectedFans) {
...
+        }
+        if (msg[4] == "FAN,LSERIES") { //this is less than perfect, but at least allows a program to continue passed the onfanfullyupdated function when a wall control is present - obviously needs to not be LSERIES specfic - need a function to just look at the "FAN" part of that string
+            	var newFan = new BigAssFan(msg[0], msg[3], address, this);
+            	this.allFans[msg[0]] = newFan;
+            	this.onFanConnection(newFan);
+            	newFan.updateAll(function() {
+                	this.onFanFullyUpdated(newFan)
+            	}.bind(this));
+            }
+        if (msg[4] == "SWITCH,SENSEME") {
```
Comment (sean9keenan) [inline BigAssApi.js]: Does this fix your #2 ?

Comment (pedroserano) [inline BigAssApi.js]: Yes although I'm sorry the fix I created is not very elegant. Let me try the split code you suggested and make other fixes. Should I request a pull again?

File: BigAssApi.js (hunk 3 above)
Comment (pedroserano) [inline BigAssApi.js]: that would be great

File: BigAssApi.js (hunk 7 above)
Comment (pedroserano) [inline BigAssApi.js:203]: Very true, or I might have confused this, could there be both a current and actual? 

[EDIT] I tried CURR again for fan speed and it came back (Haiku Fan;ERROR;PARSE). Maybe it is different for the different series?

**Pull request merged**